tooling/lsp/src/requests/references.rs

```diff
@@ -170,4 +169,65 @@ mod references_tests {
             }
         );
     }
+
+    #[test]
+    async fn ignores_macro_expansions() {
+        let src = "
+        #[foo]
+        struct Fo>|<o {}
+
+        trait Trait {}
+
+        fn main() {}
+
+        comptime fn foo(s: StructDefinition) -> Quoted {
+            let t = s.as_type();
+            quote {
+                impl Trait for $t {}
+            }
+        }
+        ";
+
+        let (mut state, noir_text_document) = test_utils::init_lsp_server("document_symbol").await;
+
+        let (line, column) = src
+            .lines()
+            .enumerate()
+            .filter_map(|(line_index, line)| {
+                line.find(">|<").map(|char_index| (line_index, char_index))
+            })
+            .next()
+            .expect("Expected to find one >|< in the source code");
```

Comment:
This pattern of finding the location of `">|<"` appears quite a few times now, along with the `apply_text_edits(&src.replace(">|<", ""), &item.additional_text_edits.unwrap());` replacement. Perhaps two helper functions wouldn't be a bad idea?

Reply:
Good point, I was being lazy here. I thought it's only used in completion tests but even then there's no excuse for duplicating this. I'll update it.

Reply:
Done!